consensus/common_test.go

```diff
@@ -492,6 +524,31 @@ func ensureVote(voteCh <-chan interface{}, height int64, round int,
 	}
 }
 
+func ensureProposal(proposalCh <-chan interface{}, height int64, round int, propId types.BlockID) {
```

Comment:
What prompted the need for this if we already have ensureNewProposal?

Reply:
allows us to also assert details about the proposal (eg we want to assert *this* specific block is proposed vs *a* block is proposed). I took the pattern from `ensureNewVote` (though it isnt actually used at the moment) vs `ensureVote`

Reply:
Yep - we probably want to replace ensureNewProposal with this eventually so we're always asserting the correct value. Fine for now.